nevergrad/optimization/optimizerlib.py

```diff
@@ -217,7 +232,10 @@ def _internal_tell(self, x: ArrayLike, value: float) -> None:
         self.listy += [value]
         if len(self.listx) >= self.es.popsize:
             try:
-                self.es.tell(self.listx, self.listy)
+                if self._fcmaes:
+                    self.es.tell(self.listy, self.listx)
```

Comment:
why is most of the code nearly the same except here? :s

Reply:
Its a design decision on fcmaes side. To process self.listx in tell costs time (involves
matrix operations)  and is unnecessary if you are "telling" the values for arguments
you obtained by ask. So self.listx is optional for fcmaes.tell, which means it has to be the second argument, not the first, since self.es.tell(self.listy) is also a valid call. 
Original cma is not designed for optimal performance, therefore it can use 
self.es.tell(self.listx, self.listy).